src/rules.h

```diff
@@ -29,6 +29,7 @@ struct rule {
 
         /* modifying */
         gint64 timeout; // this has to be the first modifying rule
+        gint64 override_dbus_timeout;
```

Comment:
You can then make this an `override_timeout` variable (set by the `override_dbus_timeout` setting). That can be used in `notification_init` after the rule processing to override the `timeout`.